### PR TITLE
Update test target to current Ubuntu

### PR DESCRIPTION
**Pull request by wiredfool**

### CODE REVIEW

File: manylinux2014-wheel-build/Makefile [hunk 1]
```diff
@@ -27,10 +28,12 @@ wheel:
 	_PYVER=39 $(MAKE) wheel
 310:
 	_PYVER=310 $(MAKE) wheel
+311:
+	_PYVER=310 $(MAKE) wheel
```
Comment (nulano) [inline manylinux2014-wheel-build/Makefile:32]:
```suggestion
311:
	_PYVER=311 $(MAKE) wheel
```
?

**Comment by radarhere** [comment]
Not quite related question - when I run this branch in my own fork, it fails, saying that it is "[Unable to find image 'githubactions/ubuntu-22.04-jammy-amd64:main' locally](https://github.com/radarhere/docker-images/actions/runs/3389569892/jobs/5632803499#step:6:753)" (whereas [main passes](https://github.com/radarhere/docker-images/actions/runs/3367721625/jobs/5585472943)).

Any ideas why that might be?

File: manylinux2014-wheel-build/build.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ fi
 # not strictly necessary, unless running multiple versions from the shell
 rm -f /tmp/*.whl || true
 
-# Python version, as 37,38,39,310. Defaults to 38.
+# Python version, as 37,38,39,310,311. Defaults to 38.
 # Matches the naming in /opt/python/
 PYVER=${1:-38}
```
Comment (radarhere) [inline manylinux2014-wheel-build/build.sh:21]:
```suggestion
# Python version, as 37,38,39,310,311. Defaults to 310.
# Matches the naming in /opt/python/
PYVER=${1:-310}
```
I think the default build Python version and the tested Python should be in sync?

**Comment by wiredfool** [comment]
There's a race condition between this image and the test image -- if we're not on main and there isn't a version of it pushed to dockerhub. Because of the branch and username, we're specifically using the test version that's built in this (or your) pr. It might get better on subsequent runs if it's actually pushing to dockerhub, or it won't.

**Comment by hugovk** [comment]
Should this target Python 3.11 by default?

**Comment by wiredfool** [comment]
No, because Jammy doesn't.

**Comment by wiredfool** [comment]
guess I'm backing up the CI queue today.

**Comment by wiredfool** [comment]
Now I'm just feeling bad for the CO2 burned:
image


Why is this attempting to run 8 times?

**Comment by radarhere** [comment]
I don't know where that screenshot comes from, but I really don't think that all of the jobs ran 8 times -  I expect it was only one of them that had 8 attempts.

Our test suite has some intermittent failures, but the main culprit is [`test_redos` in test_file_pdf.py](https://github.com/python-pillow/Pillow/blob/86b4cb6d14553d93b65ca6fbd76c433ae515aeeb/Tests/test_file_pdf.py#L288-L296). It has a timeout that tends to fail on the slower valgrind job. I tend to rerun the failing job until it passes.

**Pull request merged**